### PR TITLE
Added CRUD to Employee

### PR DESCRIPTION
**Pull request by Tivgres**
I added CRUD for the employee.

### CODE REVIEW

File: app/controllers/employees_controller.rb [hunk 1]
```diff
@@ -1,3 +1,58 @@
 class EmployeesController < ApplicationController
-  def index; end
-end
+  before_action :employee_params, only: %i[create update]
```
Comment (drkmen) [inline app/controllers/employees_controller.rb]: For what is this? You call it again in those methods.

File: app/controllers/employees_controller.rb [hunk 2]
```diff
@@ -1,3 +1,58 @@
...
+  def show
+    @employee = Employee.find(params[:id])
+  end
+
+  def new
+    @employee = Employee.new
+  end
+
+  def create
+    @employee = Employee.new employee_params
```
Comment (drkmen) [inline app/controllers/employees_controller.rb]: here (see first comment)

File: app/controllers/employees_controller.rb [hunk 3]
```diff
@@ -1,3 +1,58 @@
...
+    end
+  end
+
+  def edit
+    @employee = Employee.find(params[:id])
+  end
+
+  def update
+    @employee = Employee.find(params[:id])
+    if @employee.update(employee_params)
```
Comment (drkmen) [inline app/controllers/employees_controller.rb]: and here (see first comment)

File: app/controllers/employees_controller.rb [hunk 4]
```diff
@@ -1,3 +1,58 @@
...
-  def index; end
-end
+  before_action :employee_params, only: %i[create update]
+
+  def index
+    @employees = Employee.all
+  end
+
+  def show
+    @employee = Employee.find(params[:id])
```
Comment (drkmen) [inline app/controllers/employees_controller.rb]: But `Employee.find(params[:id])` should be in `before_action`

File: app/controllers/employees_controller.rb [hunk 5]
```diff
@@ -1,3 +1,58 @@
...
+      flash[:notice] = 'Successfully created'
+      redirect_to employee_path(@employee)
+    else
+      flash[:danger] = 'Is not created'
+      render :new
+    end
+  end
+
+  def edit
+    @employee = Employee.find(params[:id])
```
Comment (drkmen) [inline app/controllers/employees_controller.rb]: use `before_action`

File: app/controllers/employees_controller.rb [hunk 6]
```diff
@@ -1,3 +1,58 @@
...
+  end
+
+  def update
+    @employee = Employee.find(params[:id])
```
Comment (drkmen) [inline app/controllers/employees_controller.rb]: use `before_action`

File: app/controllers/employees_controller.rb [hunk 7]
```diff
@@ -1,3 +1,58 @@
...
+      flash[:notice] = 'Successfully updated'
+      redirect_to employee_path(@employee)
+    else
+      flash[:danger] = 'Is not updated'
+      render :edit
+    end
+  end
+
+  def destroy
+    @employee = Employee.find(params[:id])
```
Comment (drkmen) [inline app/controllers/employees_controller.rb]: use `before_action`

File: spec/controllers/employees_controller_spec.rb [hunk 8]
```diff
@@ -1,5 +1,93 @@
...
+    end
+
+    it 'returns a fail response and render #new' do
+      expect do
+        post :create, params: {
+          employee: { first_name: 'qwe-qwe', last_name: 'qwe-qwe', email: 'email@example.com' }
+        }
+      end.to(change { Employee.count }.by(0)) &&
+        redirect_to(Employee.last &&
+        have_http_status(200)) &&
```
Comment (Capitalsav) [inline spec/controllers/employees_controller_spec.rb]: > returns a fail response

but why `have_http_status(200)` ?

**Comment by drkmen** [comment]
Also, set correct name to the PR.

**Comment by drkmen** [comment]
My notes are still actual.

**Pull request merged**